scripts/m4t/evaluate/asr_bleu.py

```diff
@@ -0,0 +1,212 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import itertools
+import logging
+import os
+
+import sacrebleu
+import torch
+import torchaudio
+import whisper
+from fairseq2.data import Collater
+from fairseq2.data.audio import AudioDecoder, WaveformToFbankConverter
+from fairseq2.memory import MemoryBlock
+from m4t_scripts.evaluate.download_data import download_datasets
+from seamless_communication.models.inference import Translator
+from seamless_communication.models.inference.translator import Modality
+from seamless_communication.models.unity import (UnitYModel, load_unity_model,
+                                                 load_unity_text_tokenizer,
+                                                 load_unity_unit_tokenizer)
+from seamless_communication.models.vocoder import Vocoder, load_vocoder_model
+from whisper_normalizer.basic import BasicTextNormalizer
+from whisper_normalizer.english import EnglishTextNormalizer
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s %(levelname)s -- %(name)s: %(message)s",
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ASRBleu:
+    def __init__(
+        self,
+        output_dir: str,
+    ):
+        self.output_dir = output_dir
+        waveforms_dir = os.path.join(output_dir, "output_waveforms")
+        os.makedirs(waveforms_dir, exist_ok=True)
+
+    def compute_asr_bleu(
+        self,
+        lang_dir: str,
+        split: str,
+        num_data_pairs: int,
+        model_name: str,
+        eval_first_pass: bool,
+        dataset: str,
+        audio_format: str,
+    ):
+        # Set device
+        if torch.cuda.is_available():
+            device = torch.device("cuda:0")
+            dtype = torch.float16
+            logger.info(f"Running inference on the GPU in {dtype}.")
+        else:
+            device = torch.device("cpu")
+            dtype = torch.float32
+            logger.info(f"Running inference on the CPU in {dtype}.")
+
+        # Download fleurs test data
+        src_lang, tgt_lang = lang_dir.split("-")
+        download_datasets([(src_lang, tgt_lang)], split, num_data_pairs, "./data")
```

Comment:
Also, curious that do we need to download full datasets each time? If not, will we have incremental downloading based on previous cached/saved `num_data_pairs` ?

Reply:
At the moment, the data we've downloaded is cached in the output directory. The download_fleurs_dataset contains parameter: save_directory, which is later passed to the Speech2SpeechFleursDatasetBuilder Class within the function. Within this class, we have the  dataset_cache_dir parameter set to the output directory.

Could you provide further guidance on how the ideal format should be?